### PR TITLE
Add "missing" behavior to some aggregations

### PR DESCRIPTION
**Pull request by romainneutron**

### CODE REVIEW

File: src/Aggregation/Range.php [hunk 1]
```diff
@@ -58,7 +58,7 @@ public function setKeyed(bool $keyed): self
     /**
      * @return $this
      *
-     * @deprecated singe version 7.1.0, use the "setKeyed()" method instead.
+     * @deprecated since version 7.1.0, use the "setKeyed()" method instead.
```
Comment (deguif) [inline src/Aggregation/Range.php:61]: Oups ;)

File: docker/docker-compose.es75.yml [hunk 2]
```diff
@@ -2,6 +2,6 @@ version: '3.4'
 
 services:
     es01:
-        image: docker.elastic.co/elasticsearch/elasticsearch-oss:7.5.1
+        image: docker.elastic.co/elasticsearch/elasticsearch-oss:7.5.2
```
Comment (deguif) [inline docker/docker-compose.es75.yml:5]: Can you explain a little more why this version bump is needed?

File: src/Aggregation/Cardinality.php [hunk 3]
```diff
@@ -9,6 +9,7 @@
  */
 class Cardinality extends AbstractSimpleAggregation
 {
+    use MissingTrait;
```
Comment (deguif) [inline src/Aggregation/Cardinality.php:12]: CS, but I would in favor to add a line break to clearly separate from following constant declaration.

File: CHANGELOG.md [hunk 4]
```diff
@@ -23,6 +23,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+* Added `Elastica\Aggregation\Cardinality::setMissing` [#1876](https://github.com/ruflin/Elastica/pull/1876)
+* Added `Elastica\Aggregation\DateRange::setMissing` [#1876](https://github.com/ruflin/Elastica/pull/1876)
+* Added `Elastica\Aggregation\DateHistogram::setMissing` [#1876](https://github.com/ruflin/Elastica/pull/1876)
+* Added `Elastica\Aggregation\ExtendedStats::setMissing` [#1876](https://github.com/ruflin/Elastica/pull/1876)
+* Added `Elastica\Aggregation\Histogram::setMissing` [#1876](https://github.com/ruflin/Elastica/pull/1876)
+* Added `Elastica\Aggregation\Max::setMissing` [#1876](https://github.com/ruflin/Elastica/pull/1876)
+* Added `Elastica\Aggregation\Min::setMissing` [#1876](https://github.com/ruflin/Elastica/pull/1876)
+* Added `Elastica\Aggregation\Stats::setMissing` [#1876](https://github.com/ruflin/Elastica/pull/1876)
+* Added `Elastica\Aggregation\Sum::setMissing` [#1876](https://github.com/ruflin/Elastica/pull/1876)
+* Added `Elastica\Aggregation\Terms::setMissing` [#1876](https://github.com/ruflin/Elastica/pull/1876)
```
Comment (deguif) [inline CHANGELOG.md:36]:
```suggestion
* Added `Elastica\Aggregation\Avg::setMissing()` [#1876](https://github.com/ruflin/Elastica/pull/1876)
* Added `Elastica\Aggregation\Cardinality::setMissing()` [#1876](https://github.com/ruflin/Elastica/pull/1876)
* Added `Elastica\Aggregation\DateRange::setMissing()` [#1876](https://github.com/ruflin/Elastica/pull/1876)
* Added `Elastica\Aggregation\DateHistogram::setMissing()` [#1876](https://github.com/ruflin/Elastica/pull/1876)
* Added `Elastica\Aggregation\ExtendedStats::setMissing()` [#1876](https://github.com/ruflin/Elastica/pull/1876)
* Added `Elastica\Aggregation\Histogram::setMissing` [#1876](https://github.com/ruflin/Elastica/pull/1876)
* Added `Elastica\Aggregation\Max::setMissing()` [#1876](https://github.com/ruflin/Elastica/pull/1876)
* Added `Elastica\Aggregation\Min::setMissing()` [#1876](https://github.com/ruflin/Elastica/pull/1876)
* Added `Elastica\Aggregation\Stats::setMissing()` [#1876](https://github.com/ruflin/Elastica/pull/1876)
* Added `Elastica\Aggregation\Sum::setMissing()` [#1876](https://github.com/ruflin/Elastica/pull/1876)
* Added `Elastica\Aggregation\Terms::setMissing()` [#1876](https://github.com/ruflin/Elastica/pull/1876)
```

File: src/Aggregation/Percentiles.php [hunk 5]
```diff
@@ -79,15 +81,4 @@ public function addPercent(float $percent): self
...
         return $this->addParam('percents', $percent);
     }
-
-    /**
-     * Defines how documents that are missing a value should
-     * be treated.
-     *
-     * @return $this
-     */
-    public function setMissing(float $missing): self
```
Comment (deguif) [inline src/Aggregation/Percentiles.php:89]: Same than https://github.com/ruflin/Elastica/pull/1876#discussion_r528979016
See https://3v4l.org/QspR2

As a BC break should be added to the corresponding section in the changelog.

**Review by deguif — COMMENTED** [review]
Another 👍 for using trait.

**Comment by romainneutron** [comment]
PR updated comments addressed

File: CHANGELOG.md [hunk 6]
```diff
@@ -23,6 +24,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * Added `static_lambda` CS rule [#1870](https://github.com/ruflin/Elastica/pull/1870)
 * Added `Elastica\Aggregation\DateRange::setTimezone()` [#1847](https://github.com/ruflin/Elastica/pull/1847)
 * Added endpoint options support to `Elastica\Index::create()` [#1859](https://github.com/ruflin/Elastica/pull/1859)
+* Added `Elastica\Aggregation\Avg::setMissing()` [#1876](https://github.com/ruflin/Elastica/pull/1876)
```
Comment (ruflin) [inline CHANGELOG.md:27]: Seems to be a few duplicates :-)

Comment (romainneutron) [reply CHANGELOG.md]: I think it's more readable this way.
as you wish, I've turned this into a oneliner

**Review by deguif — APPROVED** [review]

File: CHANGELOG.md (hunk 6 above)
Comment (ruflin) [reply CHANGELOG.md]: 🤦‍♂️  I really thought it is duplicates but I now realised it wasn't. Sorry, I agree your version before was better :-(

**Review by ruflin — APPROVED** [review]
LGTM. Sorry about the changelog. I leave it to you which option you prefer.

**Comment by deguif** [comment]
Some conflicts to resolve, and we should be able to merge it.

**Comment by romainneutron** [comment]
PR rebased

**Comment by deguif** [comment]
It seems we have issues with Travis (too much credits consumed)
ping @ruflin 

Screenshot 2020-11-25 at 19 54 35

**Comment by ruflin** [comment]
:-( My initial instinct was to quickly buy some additional credits but seems the only option is a monthly subscription. Perhaps time to move to Github actions? Anyone experience with it? I only skimmed through https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/migrating-from-travis-ci-to-github-actions but at least it seems doable. We could try perhaps first the linting check and then move on to the other builds?

**Comment by deguif** [comment]
@romainneutron sorry for the delay but we had to migrate tests on github action.
Now that is done, could you rebase your PR so that we can merge it.

File: src/Aggregation/MissingTrait.php [hunk 7]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Elastica\Aggregation;
```
Comment (deguif) [inline src/Aggregation/MissingTrait.php:3]: Namespace was changed recently. See #1888

**Comment by romainneutron** [comment]
PR rebased

File: CHANGELOG.md [hunk 8]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Backward Compatibility Breaks
 * Allow the Terms query to accept arrays of strings, ints and floats [#1872](https://github.com/ruflin/Elastica/pull/1872)
 * Added a default value to `Elastica\Aggregation\Range::setKeyed()` and `Elastica\Aggregation\PercentilesBucket::setKeyed()` [#1876](https://github.com/ruflin/Elastica/pull/1876)
+* Added a default value to `Elastica\Aggregation\Percentiles::setMissing()` [#1875](https://github.com/ruflin/Elastica/pull/1875)
```
Comment (deguif) [inline CHANGELOG.md:11]: Missed this one, but seems the changelog for BC break is not accurate.
No default value was added but type-hint changed.

Comment (romainneutron) [reply CHANGELOG.md]: fixed

**Review by deguif — APPROVED** [review]

**Pull request merged**